pkgs/development/compilers/llvm/common/clang/default.nix

```diff
@@ -79,17 +77,12 @@ let
       # Make sure clang passes the correct location of libLTO to ld64
       substituteInPlace lib/Driver/ToolChains/Darwin.cpp \
         --replace-fail 'StringRef P = llvm::sys::path::parent_path(D.Dir);' 'StringRef P = "${lib.getLib libllvm}";'
-    '' + (
-      # See the comment on the `add-nostdlibinc-flag.patch` patch in
-      # `../default.nix` for why we skip Darwin here.
-      if lib.versionOlder release_version "13" && (!stdenv.hostPlatform.isDarwin || !stdenv.targetPlatform.isDarwin) then ''
-        sed -i -e 's/DriverArgs.hasArg(options::OPT_nostdlibinc)/true/' \
-               -e 's/Args.hasArg(options::OPT_nostdlibinc)/true/' \
-               lib/Driver/ToolChains/*.cpp
-      '' else ''
-        (cd tools && ln -s ../../clang-tools-extra extra)
-      ''
-    ) + lib.optionalString stdenv.hostPlatform.isMusl ''
+      (cd tools && ln -s ../../clang-tools-extra extra)
+    '' + lib.optionalString (lib.versionOlder release_version "13") ''
+      substituteInPlace tools/extra/clangd/quality/CompletionModel.cmake \
+        --replace ' ''${CMAKE_SOURCE_DIR}/../clang-tools-extra' ' ''${CMAKE_SOURCE_DIR}/tools/extra'
```

Comment:
@sternenseemann 

https://github.com/NixOS/nixpkgs/pull/356120 (--replace-fail) broke this one on clang 12 (staging 03ed65c58212fded6339d4e59167d2d8765ae0d5).

The path tools/extra doesn't exist; the only reference to it I could find is in the deleted diff hunk which appears after this one. I take it you somehow separated out clang-tools-extra from this derivation and therefore it is vestigial and can be removed. Correct?

Reply:
Oh no, wait, I see `(cd tools && ln -s ../../clang-tools-extra extra)` reatined in this diff which I missed. Hmm.

Reply:
Found it, it's vestigial because the patch which removed CMAKE_SOURCE_DIR was removed in 12.0.1:
https://github.com/llvm/llvm-project/commit/3be5dbbc32dd348dddcec9e3baec96b7dcef8a35

Reply:
Can't comment much more, I've just tried to preserve all the parts of the phases in the original LLVM 12 expressions. They may have been unnecessarily copied from earlier versions, yes.

Reply:
Fix for llvm 12: https://github.com/NixOS/nixpkgs/pull/367056